extension/lib/magento/customer/getDiff.js

```diff
@@ -0,0 +1,37 @@
+const { differenceWith, isEqual } = require('lodash')
+/**
+ * @param {StepContext} context
+ * @param {Object} input
+ * @param {array} input.productIds - products to put
+ * @param {array} input.magentoProductIds - products currently on the magento fav list
+ * @returns {Promise<{productIds: Object}>}
+ */
+module.exports = async (context, input) => {
+  const { favoriteList, magentoProducts } = input
+  let favIds = []
+  let existIds = []
+
+  favoriteList.map(item => {
+    favIds.push(item.id)
+  })
+  magentoProducts.map(item => {
+    existIds.push(item.id)
+  })
+
+  const removeProductIds = differenceWith(existIds, favIds, isEqual)
+  const newProductIds = differenceWith(favIds, existIds, isEqual)
+  let addProducts = []
+
+  favoriteList.map(product => {
+    newProductIds.map(id => {
+      if (id === product.id) {
+        addProducts.push(product)
+      }
+    })
+  })
```

Comment:
weird usage of map, map is a function that you use the return of. Here you use it as a foreach loop :D 
Maybe you can simplify,
You are trying to filter out the results you need, think of it as doing a `foreach` loop, and the second thing as a check for `in_array()`
I think this should work:
```
const addProducts = favoriteList.filter(product => newProductIds.includes(product.id))
```
Maybe returning the includes will help if it doesn't like shorthand.